frontend/libs/studio-components-legacy/src/components/StudioCodelistEditor/StudioCodeListEditorReducer.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { ObjectUtils } from '@studio/pure-functions';
+import { codeListWithTextResources } from './test-data/codeListWithTextResources';
+import { textResources } from './test-data/textResources';
+import { reducer, ReducerActionType } from './StudioCodeListEditorReducer';
+import type { ReducerAction, ReducerState } from './StudioCodeListEditorReducer';
+import { CodeListItemTextProperty } from './types/CodeListItemTextProperty';
+import type { TextResource } from '../../types/TextResource';
+
+// Test data:
+const defaultState = {
+  codeList: codeListWithTextResources,
+  textResources: textResources,
+};
+
+const dispatch = (action: ReducerAction, state?: Partial<ReducerState>) => {
+  return reducer(state ? { ...defaultState, ...state } : { ...defaultState }, action);
+};
+
+describe('StudioCodeListEditorReducer', () => {
+  afterEach(jest.clearAllMocks);
+
+  describe('SetCodeList', () => {
+    it('should update state correctly', () => {
+      const testCodeList = [{ value: 'test1', label: 'test2' }];
+      const expectedState = { codeList: testCodeList, textResources: textResources };
+
+      const state: ReducerState = dispatch({
+        type: ReducerActionType.SetCodeList,
+        codeList: testCodeList,
+      });
+
+      expect(state).toEqual(expectedState);
+    });
+  });
+
+  describe('AddTextResource', () => {
+    it('should update state correctly when adding text resource for label', () => {
+      const textResource: TextResource = {
+        id: 'newId',
+        value: 'some test value',
+      };
+      const codeItemIndex = 1;
+      const property = CodeListItemTextProperty.Label;
+
+      const expectedState = ObjectUtils.deepCopy(defaultState);
+      expectedState.codeList[codeItemIndex].label = textResource.id;
+      expectedState.textResources.push(textResource);
+
+      const state: ReducerState = dispatch({
+        type: ReducerActionType.AddTextResource,
+        textResource: textResource,
+        codeItemIndex: codeItemIndex,
+        property: property,
+      });
+
+      expect(state).toEqual(expectedState);
```

Comment:
Småpirk: Dette er ikke feil, men her har du egentlig laget `expectedState` litt på samme måte som i funksjonen du tester. Hva tenker du om å bare sjekke verdiene direkte? Da får du også utelatt irrelevante ting som plasseringen av den nye tekstressursen i listen.
```suggestion
      expect(state.codeList[codeItemIndex].label).toEqual(textResource.id);
      expect(state.textResources).toContain(textResource);
```
Har du for øvrig vurdert å gjøre dette til to separate tester? Én for å sjekke tekstressursen og én for å sjekke kodelisten? Da ville det blitt mer tydelig hvilken data vi ønsker at funksjonen skal påvirke.

Det samme gjelder egentlig de andre testene for oppdatering av tekstressurser.

Reply:
Jeg tenkte litt på det at det var litt rart ja. Jeg kan prøve å sette opp 2 tester for hver funkjson for å se at de oppdatere en del av state riktig